programs/castle-vault/src/instructions/consolidate_refresh.rs

```diff
@@ -88,43 +88,7 @@ pub fn handler<'info>(ctx: Context<'_, '_, '_, 'info, ConsolidateRefresh<'info>>
         let allocation: SlotTrackedValue = ctx.accounts.vault.actual_allocations[p];
         if ctx.accounts.vault.get_yield_source_availability(p) {
             // We skip pools where we have zero allocation
-            // Must do it inside the if block because we have to check yield source flags first
```

Comment:
Why remove these? The validation is still necessary

Reply:
Thinking over this again, I have a feeling that even the original check might be flawed? Need your opinion here.

We do not check if the allocation of a pool is reduced from non-zero to zero. This case eludes the check, but if somehow realized, can be used to hack the vault.

Reply:
about lines 39 to 43, yes can be deleted. and I replied on slack as well